syntax = "proto3";

service KitNet {
	rpc SayHello (HelloRequest) returns (HelloReply) {}	
}

message MacAddr {
	uint64 value = 1;
}

message IPv4Addr {
	uint32 value = 1;
}

message IpProto {
	uint32 value = 1;
}

message Port {
	uint32 value = 1;
}

message Stats {
	MacAddr mac_src = 1;
	IPv4Addr ip_src = 2;
	IPv4Addr ip_dst = 3;
	IpProto ip_proto = 4;
	Port port_src = 5;
	Port port_dst = 6;
	uint32 decay = 7;
	uint32 mac_ip_src_pkt_cnt = 8;
	uint32 mac_ip_src_mean = 9;
	uint32 mac_ip_src_std_dev = 10;
	uint32 ip_src_pkt_cnt = 11;
	uint32 ip_src_mean = 12;
	uint32 ip_src_std_dev = 13;
	uint32 ip_pkt_cnt = 14;
	uint32 ip_mean_0 = 15;
	uint32 ip_std_dev_0 = 16;
	uint32 ip_magnitude = 17;
	uint32 ip_radius = 18;
	uint32 five_t_pkt_cnt = 19;
	uint32 five_t_mean_0 = 20;
	uint32 five_t_std_dev_0 = 21;
	uint32 five_t_magnitude = 22;
	uint32 five_t_radius = 23;
	uint64 ip_sum_res_prod_cov = 24;
	uint64 ip_pcc = 25;
	uint64 five_t_sum_res_prod_cov = 26;
	uint64 five_t_pcc = 27;
}

// The request message containing the user's name.
message HelloRequest {
	string name = 1;
}

// The response message containing the greetings
message HelloReply {
	string message = 1;
}